keyboards/handwired/dactyl_manuform/6x7_fullrows/keymaps/zoullx/keymap.c

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 Zoull Danusty (@zoullx)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include QMK_KEYBOARD_H
+
+// #define RAISE MO(_RAISE)
+// #define LOWER MO(_LOWER)
+
+enum custom_layer {
+    _PDVORAK,
+    _SPDVORAK,
+    _DVORAK,
+    _QWERTY,
+    _RGBMATRIX
+    // _LOWER,
+    // _RAISE,
+};
+
+enum custom_keycodes {
+    DVORAK = DF(_DVORAK),
+    PDVORAK = DF(_PDVORAK),
+    SDVORAK = MO(_SPDVORAK),
+    QWERTY = DF(_QWERTY),
+    RGBMAT = DF(_RGBMATRIX),
+    KCSA = S(KC_A),
+    KCSB = S(KC_B),
+    KCSC = S(KC_C),
+    KCSD = S(KC_D),
+    KCSE = S(KC_E),
+    KCSF = S(KC_F),
+    KCSG = S(KC_G),
+    KCSH = S(KC_H),
+    KCSI = S(KC_I),
+    KCSJ = S(KC_J),
+    KCSK = S(KC_K),
+    KCSL = S(KC_L),
+    KCSM = S(KC_M),
+    KCSN = S(KC_N),
+    KCSO = S(KC_O),
+    KCSP = S(KC_P),
+    KCSQ = S(KC_Q),
+    KCSR = S(KC_R),
+    KCSS = S(KC_S),
+    KCST = S(KC_T),
+    KCSU = S(KC_U),
+    KCSV = S(KC_V),
+    KCSW = S(KC_W),
+    KCSX = S(KC_X),
+    KCSY = S(KC_Y),
+    KCSZ = S(KC_Z),
+    KCSTAB = S(KC_TAB),
+    KCSENT = S(KC_ENT),
+    KCSLEFT = S(KC_LEFT),
+    KCSRGHT = S(KC_RGHT),
+    KCSUP = S(KC_UP),
+    KCSDOWN = S(KC_DOWN)
+};
```

Comment:
```suggestion
#define DVORAK DF(_DVORAK)
#define PDVORAK DF(_PDVORAK)
#define SDVORAK MO(_SPDVORAK)
#define QWERTY DF(_QWERTY)
#define RGBMAT DF(_RGBMATRIX)
#define KCSA S(KC_A)
#define KCSB S(KC_B)
#define KCSC S(KC_C)
#define KCSD S(KC_D)
#define KCSE S(KC_E)
#define KCSF S(KC_F)
#define KCSG S(KC_G)
#define KCSH S(KC_H)
#define KCSI S(KC_I)
#define KCSJ S(KC_J)
#define KCSK S(KC_K)
#define KCSL S(KC_L)
#define KCSM S(KC_M)
#define KCSN S(KC_N)
#define KCSO S(KC_O)
#define KCSP S(KC_P)
#define KCSQ S(KC_Q)
#define KCSR S(KC_R)
#define KCSS S(KC_S)
#define KCST S(KC_T)
#define KCSU S(KC_U)
#define KCSV S(KC_V)
#define KCSW S(KC_W)
#define KCSX S(KC_X)
#define KCSY S(KC_Y)
#define KCSZ S(KC_Z)
#define KCSTAB S(KC_TAB)
#define KCSENT S(KC_ENT)
#define KCSLEFT S(KC_LEFT)
#define KCSRGHT S(KC_RGHT)
#define KCSUP S(KC_UP)
#define KCSDOWN S(KC_DOWN)
```

Reply:
In the documentation here: https://docs.qmk.fm/#/contributing?id=documentation
and in the keymap that I looked at to gain understanding to create my keymap here: https://github.com/qmk/qmk_firmware/blob/master/keyboards/handwired/dactyl_manuform/3x5_3/keymaps/dlford/keymap.c
enums are used and seem to be the preferred in the documentation. I'm wrong in thinking that and should still change those to defines?

Reply:
`define`'s are for creating aliases, which is what you're doing. `enum`'s are for enumerating objects. I have a feeling the confusion is from the [macro doc](https://docs.qmk.fm/#/feature_macros?id=using-macros-in-c-keymaps).